File: data/pc/cl6/uri1001.cpp
#include <iostream>

int numbers[2];
int result;

bool read_data() {
  bool flag = true;

  numbers[0] = numbers[1] = 0;
  std::cin >> numbers[0];
  if (numbers[0] == 0) return false;
  std::cin >> numbers[1];

  if (numbers[0] == 0 || numbers[1] == 0)
    flag = false;

  return flag;
}

void process_data() {
  result = numbers[1] + numbers[0];
}

void write_data() {
  std::cout << "X = " << result << '\n';
}

int main() {
  bool flag = true;

  flag = read_data();
  while (flag) {
    process_data();
    write_data();
    flag = read_data();
  }

  return 0;
}

File: data/pc/cl6/uri1001.java
import java.util.*;

public class uri1001 {
  static Scanner input = new Scanner(System.in);
  static int[] numbers = new int[2];
  static int result;

  static boolean readData() {
    boolean flag = true;

    numbers[0] = numbers[1] = 0;

    numbers[0] = input.nextInt();
    if (numbers[0] == 0) return false;
    numbers[1] = input.nextInt();

    if (numbers[0] == 0 || numbers[1] == 0) {
      flag = false;
    }

    return flag;
  }

  static void processData() {
    result = numbers[0] + numbers[1];
  }

  static void writeData() {
    System.out.println("X = " + result);
  }

  public static final void main(String[] args) {
    boolean flag = readData();

    while (flag) {
      processData();
      writeData();
      flag = readData();
    }
  }
}
